src/Parser/Core/Modules/StatTracker.js

```diff
@@ -294,24 +321,52 @@ class StatTracker extends Analyzer {
-      this._changeStats(statBuff, event.newStacks - event.oldStacks);
+
+      const before = Object.assign({}, this._stats);
+      const delta = this._changeStats(statBuff, event.newStacks - event.oldStacks);
+      const after = Object.assign({}, this._stats);
+      this._triggerChangeStats(event, before, delta, after);
+      debug && console.log(`StatTracker: (${event.oldStacks} -> ${event.newStacks}) ${SPELLS[spellId] ? SPELLS[spellId].name : spellId} - Change: ${this._statPrint(delta)}`);
+      debug && this._debugPrintStats(this._stats);
     }
   }
 
   _changeStats(change, factor) {
-    this._stats.strength += this._getBuffValue(change, change.strength) * factor;
-    this._stats.agility += this._getBuffValue(change, change.agility) * factor;
-    this._stats.intellect += this._getBuffValue(change, change.intellect) * factor;
-    this._stats.stamina += this._getBuffValue(change, change.stamina) * factor;
-    this._stats.crit += this._getBuffValue(change, change.crit) * factor;
-    this._stats.haste += this._getBuffValue(change, change.haste) * factor;
-    this._stats.mastery += this._getBuffValue(change, change.mastery) * factor;
-    this._stats.versatility += this._getBuffValue(change, change.versatility) * factor;
-    this._stats.avoidance += this._getBuffValue(change, change.avoidance) * factor;
-    this._stats.leech += this._getBuffValue(change, change.leech) * factor;
-    this._stats.speed += this._getBuffValue(change, change.speed) * factor;
-    debug && this._debugPrintStats();
+    const delta = {
+      strength: this._getBuffValue(change, change.strength) * factor,
+      agility: this._getBuffValue(change, change.agility) * factor,
+      intellect: this._getBuffValue(change, change.intellect) * factor,
+      stamina: this._getBuffValue(change, change.stamina) * factor,
+      crit: this._getBuffValue(change, change.crit) * factor,
+      haste: this._getBuffValue(change, change.haste) * factor,
+      mastery: this._getBuffValue(change, change.mastery) * factor,
+      versatility: this._getBuffValue(change, change.versatility) * factor,
+      avoidance: this._getBuffValue(change, change.avoidance) * factor,
+      leech: this._getBuffValue(change, change.leech) * factor,
+      speed: this._getBuffValue(change, change.speed) * factor,
+    };
+
+    Object.keys(this._stats).forEach(key => {
+      this._stats[key] += delta[key];
+    });
+
+    return delta;
+  }
+
+  /*
+   * Fabricates an event indicating when stats change
+   */
+  _triggerChangeStats(event, before, change, after) {
+    this.owner.triggerEvent('changestats', {
+      timestamp: event ? event.timestamp : this.owner.currentTimestamp,
+      type: 'changestats',
+      sourceID: event ? event.sourceID : this.owner.playerId,
+      targetID: this.owner.playerId,
+      reason: event,
+      before,
+      change,
```

Comment:
How about naming this `delta` also?

Reply:
Hah, I was thinking you were going to suggest making all of them 'change'. I chose `change` over `delta` for the outward facing field because I felt it was easier to understand.

Reply:
Delta is used by browser for things like this, while `change` is usually a method.

Reply:
Alright I'll change it